overlord/ifacestate/implicit.go

```diff
@@ -33,10 +33,21 @@ import (
 // It is assumed that slots have names matching the interface name. Existing
 // slots are not changed, only missing slots are added.
 func addImplicitSlots(snapInfo *snap.Info) {
-	if snapInfo.Type != snap.TypeOS {
+	// Implicit slots can be added to the special "snapd" snap or to snaps with
+	// type "os". Currently there are no other snaps that gain implicit
+	// interfaces.
+	if snapInfo.Type != snap.TypeOS && snapInfo.InstanceName() != "snapd" {
 		return
 	}
-	// Ask each interface if it wants to be implcitly added.
+
+	// If the manager has chosen to put implicit slots on the "snapd" snap
+	// then stop adding them to any other core snaps.
+	_, isSnapdHostingImplicitSlots := mapper.(*CoreSnapdSystemMapper)
```

Comment:
we could have a predicate method on mapper instead mabye

Reply:
+1